### PR TITLE
REVIEW on rework-version-control branch

### PR DESCRIPTION
**Pull request by JosephCrispell**
Hi,

Thanks so much for sharing the [**Quality Assurance of Code for Analysis and Research**](https://best-practice-and-impact.github.io/qa-of-code-guidance/intro.html) with me. I have completed a review of the content and the current pull-request contains some very minor edits and comments throughout. 

How I did the review:
- Forked the [`qa-code-guidance`]() repository to the DFID institute page [here](https://github.com/DFID/qa-of-code-guidance)
- Create the [`rework-versioncontrol-jc-review`](https://github.com/DFID/qa-of-code-guidance/tree/rework-version-control-jc-review) branch to work on
- Made minor edits throughout to correct typos
- Added html comments (`<!-- ... -->`) in the code in a few places

A few general comments:
- As a resource the book is great ❤️. There is a lot of extremely useful content providing a strong foundation for reproducible data science projects 
- You could consider splitting the "Core programming practices" into multiple sections as it is quite long. Perhaps take the "notebooks" section and everything after and put into a new section?
- The "Structuring your project" section could possibly do with a few more visuals to break up the text
- For development ease I would add numbers to the section `.md` files so they clearly map to book sections

I see that this branch can't automatically merge - I think because it has fallen behind the on-going work on the version control section but I am happy to make any changes need on my end to ease merge.

I hope the review is useful and thank for the opportunity to explore this amazing book!

### CODE REVIEW

**Comment by JosephCrispell** [comment]
HI @foster999 - I've created the pull-request without adding any reviewers or assigning myself by mistake and don't have the permissions to change now so I'm just tagging you here and I'll drop you an email.

**Comment by foster999** [comment]
Hi @JosephCrispell,

Thanks very much for taking the time to review the book! It's really great to hear that you'll be able to make use of it too. It sounds like FCDO provide similar RAP support to our team - a major motivation for the book was to provide a reference for these analysis projects.

I've done quite a bit of rearranging in the version control section, so I think that might have caused some conflicts. I'll review the suggested changes here and then it would be great if you could resolve the conflicts for merging. I agree with the general comments above and will create issues for these, in case these don't get addressed in the current round of edits.

Once merged I'll try to address inline comments in my current PR, or if they relate to planned large-scale changes I'll move these into related issues again.

P.S. I was a microbiologist in a previous life, so looking forward to checking out your genomics tools!

**Comment by JosephCrispell** [comment]
Hi @foster999,

No problem at all - it was an extremely useful exercise and I'm going to keep it as a reference that I'll keep dipping back into. Yes, definitely a big part of our hub's work in on RAP and I will be pointing mentees to your book once released. I've also developed a few RAP resources that we use in our RAP mentoring that might be relevant ([here](https://github.com/datasciencecampus/gov-uk-rap-materials)).

Yes, that'll ne no problem let me know when you're needing me to make the changes the version control section.

Thanks!

P.S Ha! Yes please do, if you spot anything interesting then give me a call and we can discuss it further. I'd be keen to hear about your microbiologist days.

**Review by foster999 — CHANGES_REQUESTED** [review]
Thanks again @JosephCrispell, these are all great improvements.

I've made a few suggested tweaks, but more than happy to merge this once they're resolved! As in the comments, please feel free to add additional R examples, but I'll make a note to fill in those gaps anyway.

Thanks for the link to the campus RAP resources, I'll include this in the references section where we link to other guidance and useful info.

File: book/core_programming.md
Comment (JosephCrispell) [reply book/core_programming.md]: Agreed, that's a typo on my part :-D

File: book/core_programming.md
Comment (JosephCrispell) [reply book/core_programming.md]: Yes, I like that change. Too often programmers think their code is reproducible (myself included) without fully considering the steps that would still be necessary for someone to understand and replicate our work **but** using a programming script rather than manual steps is a massive step in the right direction!

File: book/core_programming.md
Comment (JosephCrispell) [reply book/core_programming.md]: Just committed a slight edit to name and content of these functions.

Comment (foster999) [reply book/core_programming.md]: Thanks, that's better

**Review by foster999 — APPROVED** [review]
Happy to merge this now - let me know when you are 😄 I can merge manually and resolve the conflicts then, or we can merge here if you have chance to resolve them at your end

File: book/configuration.md
Comment (JosephCrispell) [reply book/configuration.md:35]: Hi David - I had hoped to quickly write an R code example here but time has ran away with me!

**Comment by JosephCrispell** [comment]
Hi @foster999, the conflicts are now resolved and happy for you to merge 👍

**Comment by foster999** [comment]
Thanks again for this @JosephCrispell, really valuable input! These will be included in the next version, which will be up before the end of the month.

Please do give me a shout if you have any other comments down the line. I'd be eager to hear if there are other topics that we should look to include, once the existing sections are up to scratch.

**Pull request merged**
